sympy/functions/elementary/exponential.py

```diff
@@ -436,11 +436,27 @@ def _eval_nseries(self, x, n, logx):
         if arg0 in [-oo, oo]:
             return self
         t = Dummy("t")
-        exp_series = exp(t)._taylor(t, n)
+        nterms = n + 2
+        try:
+            cf = Order(arg.as_leading_term(x), x).getn()
+        except NotImplementedError:
+            cf = 0
+        if cf != 0:
+            try:
+                nterms = (n/cf).ceiling()
+            except (AttributeError, TypeError):
+                pass
+        exp_series = exp(t)._taylor(t, n, nterms)
         o = exp_series.getO()
         exp_series = exp_series.removeO()
         r = exp(arg0)*exp_series.subs(t, arg_series - arg0)
-        r += Order(o.expr.subs(t, (arg_series - arg0)), x)
+        if cf != 0:
+            try:
+                r += Order(o.expr.subs(t, (arg_series - arg0)), x)/x**((cf-1)*n)
+            except TypeError:
+                pass
+        else:
+            r += Order(o.expr.subs(t, (arg_series - arg0)), x)
```

Comment:
Shouldn't the try block cover also this?

Reply:
Actually It is not giving any error it is just in infinite loop(what it seems). what exception will we `catch` here?
I also noticed that it is because of `nterms` if we use normal `n` it seems to work.

Reply:
It works now
I figured it out that sometimes nterms evaluated will be negative so I have added a check:
```
if ntems < 0:
    nterms = n

Reply:
Does this mean that `cf < 0`? In that case, there will be no series expansion. It seems that currently `self` is returned unevaluated. `gruntz` is sensitive to the result. That is probably what it will expect. But if `cf < 0`, then the limit `arg0` cannot be finite. Can you check its value in case `nterm` becomes negative.

Reply:
ok for the given example:

(`gruntz(exp((log(log(x + exp(log(x)*log(log(x))))))/ (log(log(log(exp(x) + x + log(x)))))), x, oo) == E`):
Value of arg0 is `log(log(_p + exp(log(_p)*log(log(_p)))))/log(log(_p))`

value of cf is `-1`

Reply:
I just noticed that on current master `gruntz(exp((log(log(x + exp(log(x)*log(log(x))))))/ (log(log(log(exp(x) + x + log(x)))))), x, oo) == E)` it is evaluated as `False`
But on sympy live it is evaluated as `True`.